examples/chapter1/annexa/src/example4.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "simple_publisher.h"
+#include <iostream>
+
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+int main()
+{
+    std::cout << "Main thread ID: " << std::this_thread::get_id() << std::endl;
+
+    kpsr::EventEmitterMiddlewareProvider<std::string> provider(nullptr,
+                                                               "tutorial_app_api_example4",
+                                                               0,
+                                                               nullptr,
+                                                               nullptr);
+
+    SimplePublisher simplePublisher(provider.getPublisher());
+    const std::string captured_message = "Hello World! I am a captured message!";
+
+    provider.getSubscriber()
+        ->registerListener("example4", [captured_message](const std::string &message) {
```

Comment:
I'm confused: by "captures a variable by const value" do you mean `message`? It seems like a `&reference`

A different thing is that pub/sub messaging copies the variable (by = or clone) and changes in the publisher do not propagate to the listener (to my current knowledge).

Reply:
The variable captured by const value is "captured_message" in this case, which is the variable inside the "[ ]" of the lambda, and by default lambda functions capture it by const value.

Reply:
To make this point more explicit, remove the `const` for `captured_message`, and add this before the call to `simplePublisher.run` on line 40: (outside the lambda)
```
captured_message = "New value of captured message";
std::cout << captured_message << std::endl;
simplePublisher.run();
```